libraries/launcher/legacy/org/prismlauncher/legacy/LegacyLauncher.java

```diff
@@ -96,7 +96,13 @@ public LegacyLauncher(Parameters params) {
 
     @Override
     public void launch() throws Throwable {
-        Class<?> main = ClassLoader.getSystemClassLoader().loadClass(mainClassName);
+        Class<?> main;
+        try {
+            main = ClassLoader.getSystemClassLoader().loadClass(mainClassName);
+        } catch (ClassNotFoundException e) {
+            main = findClientClass(ClassLoader.getSystemClassLoader().loadClass(appletClass));
```

Comment:
This function calls createAppletClass which ends up loading the class another time. I'd prefer if this was stored in a variable and the method was changed to createApplet(Class<?> appletClass) simply as it seems a bit cleaner